content/docs/setup/kubernetes/mesh-expansion/index.md

```diff
@@ -213,11 +221,41 @@ spec:
       ports:
         http: 8080
       labels:
-        app: vm1
+        app: vmhttp
         version: 1
 EOF
 {{< /text >}}
 
+1. Workloads in Kubernetes cluster need a DNS mapping to resolve the service running on the VM.
+You can integrate with your own DNS system. For illustration purpose, we use `istioctl register`
+which creates a Kubernetes `selector-less` service.
+
+    {{< text bash >}}
+    $ istioctl  register vmhttp 0.0.0.0 8080
```

Comment:
The IP address here, why is it '0.0.0.0'?

Reply:
Good catch. Changed. The actual value of this does not matter as explained in the doc, because this is just to generate a DNS entry for the vmhttp service. But 0.0.0.0 won't work.

Reply:
I'm not finding an explanation in the doc about why that parameter doesn't matter. Could you point me to it?

Also, are there cases where that `istio register` parameter does matter? If not, can we just remove that parameter from the command?

Reply:
Pilot uses ServiceEntry for service discovery to support mesh expansion. istioctl register creates a selectorless DNS, that's all we need for it. For HTTP service, we don't care DNS because we use host header to route. TCP service that might matter. I'm not aware of any doc mentioning that.

Yeah these are all arcane knowledge not well we explained throughout.

And we can improve istioctl register. I don't think is up-to-date with current mesh expansion design.

Let's focus on getting e2e working in this PR for now and address these improvements as follow-up.